2021q1 release

salkinium opened the pull request:
Let's start doing releases again.
We'll do this in a modular way and with some scripting support to generate the links and to copy all the modular changelogs into a large `CHANGELOG.md` file.

I've added the quarterly changelogs all the way back to the last xpcc release 2018q2:
- [x] 2021q1
- [x] 2020q4 
- [x] 2020q3 
- [x] 2020q2 
- [x] 2020q1 
- [x] 2019q4 
- [x] 2019q3 
- [x] 2019q2 
- [x] 2019q1  
- [x] 2018q4 
- [x] 2018q3
- [x] Script to build whole changelog

Any opinions on the format (it's from xpcc)?

Closes #44.

cc @rleh @chris-durand @strongly-typed

rleh commented on CHANGELOG.md:
```diff
@@ -0,0 +1,851 @@
+
+<!--/releases-->
+
+## Changes previous to 2020
+
+Please interpret the git history for older changes.
```
Maybe we can include the changelog from xpcc (`docs/CHANGELOG.md`) below this section?

salkinium replied:
I've added more context and linked to the old xpcc changelog as well as to the porting guide (with a fixed sha, since I've deleted it since).

rleh reviewed (approved):
👍🏽 ❤️

rleh commented:
I would like to see a link to the changelog somewhere in the README.

chris-durand reviewed (approved):
Very nice!

salkinium commented on README.md:
```diff
@@ -292,8 +289,12 @@ Fabian Greif ([\@dergraaf](https://github.com/dergraaf)),
+David Hebbeker ([\@dhebbeker](https://github.com/dhebbeker)),
+Thorsten Lajewski ([\@TheTh0r](https://github.com/TheTh0r)),
+Mike Wolfram ([\@mikewolfram](https://github.com/mikewolfram))
+and [many more contributors][contributors].
```
Continued discussion from https://github.com/modm-io/modm/pull/592#discussion_r604253579:

I've made >20 the new cutoff and linked to the GitHub contributors page. It's arbitrary, but I also don't have a better way and also I don't really want to spend too much time on it. The releases/changelog explicitly call out the contributors again and also the individual contributions in the changelog.

salkinium commented:
With some scripting and regex, we can now write the release notes in normal markdown and the links get automatically created for the CHANGELOG.md. This makes it so, so much easier to write the release notes.

The real cherry on top is that I can now simply copy-paste the release notes without *any* changes into the GitHub Release UI and it autolinks everything automagically **with inline preview** 🥰:

<details>

![](https://user-images.githubusercontent.com/163066/113038041-4ee9d000-9196-11eb-9bd7-b96973c4abaa.png)

</details>

salkinium commented:
Is there a use-case for having the commit sha available? To look something up in the Git history?

salkinium commented:
I'll merge this tonight and tag the release 2021q1.

salkinium committed:
[release] Update changelog for 2021q1 release

chris-durand committed:
[stm32] Fix interrupt loop for uart with tx buffer but no rx buffer

The RXNE interrupt was also enabled if only tx was buffered. The
controller gets stuck in an endless loop because the flag is never
acknowledged in the ISR.

salkinium committed:
[ci] Always upload all HAL job logs

salkinium committed:
[ci] Separate check for HAL support matrix
